File: public/docs/_examples/toh-6/dart/lib/hero_detail_component.dart
// #docplaster
// #docregion , v2
import 'dart:async';
import 'dart:html';

import 'package:angular2/core.dart';
import 'package:angular2/router.dart';

import 'hero.dart';
import 'hero_service.dart';

@Component(
    selector: 'my-hero-detail',
    templateUrl: 'hero_detail_component.html',
    styleUrls: const ['hero_detail_component.css']
    )
class HeroDetailComponent implements OnInit {
  Hero hero;
  final HeroService _heroService;
  final RouteParams _routeParams;

  HeroDetailComponent(this._heroService, this._routeParams);

  Future<Null> ngOnInit() async {
    var idString = _routeParams.get('id');
    var id = int.parse(idString, onError: (_) => null);
    if (id != null) hero = await (_heroService.getHero(id));
  }

  // #docregion save
  Future<Null> save() async {
    await _heroService.update(hero);
    goBack();
  }
  // #enddocregion save

  void goBack() {
    window.history.back();
  }
}

File: public/docs/_examples/toh-6/ts/src/app/hero-detail.component.ts
// #docregion
import 'rxjs/add/operator/switchMap';
import { Component, OnInit }      from '@angular/core';
import { ActivatedRoute, Params } from '@angular/router';
import { Location }               from '@angular/common';

import { Hero }        from './hero';
import { HeroService } from './hero.service';

@Component({
<<<<<<< HEAD:public/docs/_examples/toh-6/ts/src/app/hero-detail.component.ts
  selector: 'my-hero-detail',
=======
  moduleId: module.id,
  selector: 'hero-detail',
>>>>>>> docs(toh-3): heavy copy edit; my-hero-detail -> hero-detail:public/docs/_examples/toh-6/ts/app/hero-detail.component.ts
  templateUrl: './hero-detail.component.html',
  styleUrls: [ './hero-detail.component.css' ]
})
export class HeroDetailComponent implements OnInit {
  hero: Hero;

  constructor(
    private heroService: HeroService,
    private route: ActivatedRoute,
    private location: Location
  ) {}

  ngOnInit(): void {
    this.route.params
      .switchMap((params: Params) => this.heroService.getHero(+params['id']))
      .subscribe(hero => this.hero = hero);
  }

  // #docregion save
  save(): void {
    this.heroService.update(this.hero)
      .then(() => this.goBack());
  }
  // #enddocregion save

  goBack(): void {
    this.location.back();
  }
}
